microsoft/testsuites/dpdk/testpmd.py

```diff
@@ -0,0 +1,155 @@
+            if isinstance(self.node, _os):
+                return True
+        return False
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git, Wget]
+
+    def _install(self) -> bool:
+        if self._install_dependencies():
+            node = self.node
+            git_tool = node.tools[Git]
+            git_tool.clone(self._dpdk_github, cwd=node.working_path)
+            dpdk_path = self.node.working_path.joinpath("dpdk")
+            self.__execute_assert_zero("meson build", dpdk_path)
+            dpdk_build_path = dpdk_path.joinpath("build")
+            self.__execute_assert_zero("which ninja", dpdk_build_path)
+            self.__execute_assert_zero("ninja", dpdk_build_path)
+            self.__execute_assert_zero("ninja install", dpdk_build_path)
+            self.__execute_assert_zero("ldconfig", dpdk_build_path)
+            return True
+        else:
+            return False
+
+    def _install_dependencies(self) -> bool:
+        node = self.node
+        cwd = node.working_path
+        if isinstance(node.os, Ubuntu):
+            for package in self._ubuntu_packages:
+                node.os.install_packages(package)
+            self.__execute_assert_zero("pip3 install --upgrade meson", cwd)
+            self.__execute_assert_zero("mv /usr/bin/meson /usr/bin/meson.bak", cwd)
+            self.__execute_assert_zero("ln -s /usr/local/bin/meson /usr/bin/meson", cwd)
+            self.__execute_assert_zero("pip3 install --upgrade ninja", cwd)
+            self.__execute_assert_zero(
+                "modprobe -a rdma_cm rdma_ucm ib_core ib_uverbs mlx4_core mlx4_ib", cwd
+            )
+
+        elif isinstance(node.os, Redhat) or isinstance(node.os, CentOs):
+            self.__execute_assert_zero(
+                "yum update -y --disablerepo='*' --enablerepo='*microsoft*'", cwd
+            )
+            node.execute(
+                "yum groupinstall -y 'Infiniband Support'",
+                sudo=True,
+                shell=True,
+            )
+            for package in self._redhat_packages:
+                node.os.install_packages(package)
+            self.__execute_assert_zero(
+                "modprobe -a ib_core ib_uverbs mlx4_en mlx4_core mlx5_core mlx4_ib mlx5_ib",
+                cwd,
+            )  # add mellanox drivers
+            self.__execute_assert_zero("systemctl enable rdma", cwd)
+            self.__execute_assert_zero("pip3 install --upgrade meson", cwd)
+            self.__execute_assert_zero("ln -s /usr/local/bin/meson /usr/bin/meson", cwd)
+
+            wget_tool = self.node.tools[Wget]
+            wget_tool.get(self._ninja_url)
+            node.execute(f"mv ninja-linux.zip {node.working_path}/")
+            self.__execute_assert_zero(
+                "unzip ninja-linux.zip && mv ninja /usr/bin/ninja", cwd
+            )
+            self.__execute_assert_zero("pip3 install --upgrade pyelftools", cwd)
+        else:
+            return False
```

Comment:
It's better to raise an exception instead of returning false. If the OS is not supported, either skip or fail the test. False has no enough information for troubleshooting.